src/server/shell/surface_specification.cpp

```diff
@@ -147,6 +158,66 @@ void msh::SurfaceSpecification::update_from(SurfaceSpecification const& that)
         exclusive_rect = that.exclusive_rect;
     if (that.application_id.is_set())
         application_id = that.application_id;
+    if (that.server_side_decorated.is_set())
+        server_side_decorated = that.server_side_decorated;
     if (that.focus_mode.is_set())
         focus_mode = that.focus_mode;
 }
+
+bool msh::operator==(
+    const msh::SurfaceSpecification& lhs,
+    const msh::SurfaceSpecification& rhs)
+{
+    return
+        lhs.name == rhs.name &&
+        lhs.width == rhs.width &&
+        lhs.height == rhs.height &&
+        lhs.top_left == rhs.top_left &&
+        lhs.buffer_usage == rhs.buffer_usage &&
+        lhs.pixel_format == rhs.pixel_format &&
+        lhs.input_mode == rhs.input_mode &&
```

Comment:
This is also mentioned on line 204. (It is really annoying there's no easy way to write `return std::tie($all_data_members) == std::tie(rhs.$all_data_members);`

Reply:
Fixed